### PR TITLE
remove systype and attendant platform detection.

### PR DESCRIPTION
**Pull request by alanpost**
It's used in one place, make-makelib, to exclude ranlib on
platforms that no longer exist.

### CODE REVIEW

**Comment by DerDakon** [comment]
It would be cool if someone could tests this on a recent hpux system, all others are dead.

**Comment by schmonz** [comment]
> It would be cool if someone could tests this on a recent hpux system, all others are dead.

According to this [HP-UX 11i ranlib(1) manpage](https://nixdoc.net/man-pages/HP-UX/man1/ranlib.1.html), it has `ranlib` these days.

File: Makefile [hunk 1]
```diff
@@ -361,8 +361,8 @@ case.h
 	./compile commands.c
 
 compile: \
-make-compile warn-auto.sh systype
-	( cat warn-auto.sh; ./make-compile "`cat systype`" ) > \
```
Comment (schmonz) [inline Makefile]: Was `systype` even being used in any way by `./make-compile` and `./make-load`? I can't see that it was.

Comment (schmonz) [reply Makefile]: Oh, you explained this in the very first comment here.

**Pull request merged**